osu.Game/Rulesets/Scoring/HealthProcessor.cs

```diff
@@ -92,7 +101,10 @@ private bool meetsAnyFailCondition(JudgementResult result)
                 {
                     bool conditionResult = (bool)condition.Method.Invoke(condition.Target, new object[] { this, result })!;
                     if (conditionResult)
+                    {
+                        FailTrigger = condition.Target;
                         return true;
+                    }
```

Comment:
This feels flaky. What is `FailTrigger` supposed to be? It doesn't seem like it can be guaranteed to be a mod, and based on however way `FailConditions` is being fed by the mod implementation, it might be that `condition.Target != the mod itself`, breaking apart everything in this solution.

Can we make it so that each mod explicitly provide their fail conditions in the form of a bool-returning method as part of an interface such as `IHasFailConditions` and iterate through those in here instead of having each mod subscribe to `FailConditions`?

Reply:
Alright i will see what i can do about that